pandas/io/parsers/readers.py

```diff
@@ -545,12 +549,26 @@ class _read_shared(TypedDict, Generic[HashableT], total=False):
 ...     parse_dates=[1, 2],
 ...     date_format={{'col 2': '%d/%m/%Y', 'col 3': '%a %d %b %Y'}},
 ... )  # doctest: +SKIP
-
 >>> df.dtypes  # doctest: +SKIP
 col 1             int64
 col 2    datetime64[ns]
 col 3    datetime64[ns]
 dtype: object
+
+The csv in the pyarrow must be used if the values in the file have
+new line characters.
+
+>>> from pyarrow import csv
+>>> parse_options = csv.ParseOptions(newlines_in_values=True)
+>>> table = csv.read_csv("example.csv", parse_options=parse_options)
```

Comment:
```suggestion
>>> import io
>>> from pyarrow import csv
>>> rows = [{"text": "ab\ncd", "idx": idx} for idx in range(1_000_000)]
>>> df = pd.DataFrame(rows)
>>> source = io.BytesIO(df.to_string(index=False).encode())
>>> parse_options = csv.ParseOptions(newlines_in_values=True)
>>> table = csv.read_csv(source, parse_options=parse_options)
```
The doctest will fail because "example.csv" does not exist. We can change a bit or just skip the test.

Reply:
Instead of changing the example codes, I tried to skip the doctest because most of use cases with pyarrow are probably using csv files.

Reply:
I am pretty new to numpy contribution. If the example codes need to be modified as suggested, please let me know.